src/Database/Type/DateTimeFractionalType.php

```diff
@@ -0,0 +1,40 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+/**
+ * Extends DateTimeType with support for fractional seconds up to microseconds.
+ */
+class DateTimeFractionalType extends DateTimeType
```

Comment:
Wouldn't  DateTimeMicrosecondType be a better name? Current name sounds quite odd to me.

Reply:
I agree it is a little awkward.  I was trying to do a few things:

1. Keep the name aligned with the various standards language.  
2. Keep 'DateTime' as the prefix to read easier
3. Allow TimeFractionalType later
4. Allow the new types to line up with new type map names: 'datetime??', 'timestamp??'

You can specify fractional seconds kept in the various datetime/timestamp data types.  Both mysql and postgresql allow you to specify 0-6.

The idea is you can actually have a field that stores milliseconds or microseconds or a single digit.  Fractional seconds describes any amount of precision which means it works with any of these type declarations.

If we have DateTimeMicrosecondsType, it could imply that it only works with DATETIME(6) and TIMESTAMPTZ(6) types.

Reply:
Let me know what you want to name it, and I'll change it.  Keep in mind that DateTimeTimezoneType will come after.

Reply:
NVM, existing class name is fine.

Reply:
The alternative is DateTimeWithFractionalSecondsType and TimeWithFractionalSecondsType.

I went with slightly awkward over extremely verbose lol.